src/EventStore.Client/EventStoreClientBase.cs

```diff
@@ -32,38 +38,51 @@ public abstract class EventStoreClientBase : IDisposable, IAsyncDisposable {
 		/// <param name="exceptionMap"></param>
 		protected EventStoreClientBase(EventStoreClientSettings? settings,
 			IDictionary<string, Func<RpcException, Exception>> exceptionMap) {
-			_exceptionMap = exceptionMap;
 			Settings = settings ?? new EventStoreClientSettings();
+			_exceptionMap = exceptionMap;
+			_cts = new CancellationTokenSource();
+			_channelSelector = new ChannelSelector(Settings, _cts.Token);
+			_channelInfoLazy = new Lazy<Task<ChannelInfo>>(() => _channelSelector.SelectChannel(_cts.Token),
+				LazyThreadSafetyMode.PublicationOnly);
```

Comment:
if two threads execute the lazy at the same time, i think one will succeed but the other will wait and consume the _next_ channel selection if it is a gossipchannelselector